pkg/internal/pipe/instrumenter.go

```diff
@@ -51,17 +49,46 @@ func Build(ctx context.Context, config *beyla.Config, ctxInfo *global.ContextInf
 		}, nil
 	})
 
-	// process subpipeline optionally starts another pipeline only to collect and export data
-	// about the processes of an instrumented application
-	swi.Add(ProcessMetricsSwarmInstancer(ctxInfo, config, ctxInfo.OverrideAppExportQueue))
-
 	selectorCfg := &attributes.SelectorConfig{
 		SelectionCfg:            config.Attributes.Select,
 		ExtraGroupAttributesCfg: config.Attributes.ExtraGroupAttributes,
 	}
-
 	swi.Add(alloy.TracesReceiver(ctxInfo, &config.TracesReceiver, config.Metrics.SpanMetricsEnabled(),
 		selectorCfg, ctxInfo.OverrideAppExportQueue))
 
+	clusterConnectorsSubpipeline(swi, ctxInfo, config)
+
+	swi.Add(ProcessMetricsSwarmInstancer(ctxInfo, config, ctxInfo.OverrideAppExportQueue))
+
 	return swi.Instance(ctx)
 }
+
+// clusterConnectorsSubpipeline will submit "connector" traces that are identified as cluster-external.
+// Tempo will use them to compose inter-cluster service graph connections that otherwise couldn't be composed by
+// Beyla, as it lacks the metadata from the remote clusters.
+func clusterConnectorsSubpipeline(swi *swarm.Instancer, ctxInfo *global.ContextInfo, config *beyla.Config) {
+	if !slices.Contains(config.Topology.Spans, spanscfg.TopologyInterCluster) {
+		return
+	}
+
+	externalTraces := msg.NewQueue[[]request.Span](msg.ChannelBufferLen(config.ChannelBufferLen))
+	swi.Add(traces.SelectExternal(
+		ctxInfo.OverrideAppExportQueue,
+		externalTraces,
+	))
+
+	swi.Add(alloy.ConnectionSpansReceiver(ctxInfo,
+		&beyla.TracesReceiverConfig{
+			Traces:           config.TracesReceiver.Traces,
+			Instrumentations: config.TracesReceiver.Instrumentations,
+			Sampler: services.SamplerConfig{
+				Name: "always_off", // parentbased_always_on?
```

Comment:
Does `always_off` mean that the connection spans are not actually exported? The unit tests are using `always_on`.